runtime/auth/http-auth-aws/common/src/aws/smithy/kotlin/runtime/http/auth/AwsHttpSigner.kt

```diff
@@ -134,6 +140,7 @@ public class AwsHttpSigner(private val config: Config) : HttpSigner {
             algorithm = config.algorithm
 
             region = when {
+                algorithm == AwsSigningAlgorithm.SIGV4_ASYMMETRIC && config.clientSigv4aSigningRegionSet != null -> config.clientSigv4aSigningRegionSet!!.toSet().joinToString(",")
```

Comment:
question: I noticed we call `.toSet()`, why not just make it a `Set` instead of a `List` in configuration?

Reply:
I'd err on the side of leaving it a list should order ever matter.

**correctness**: This is favoring the value set on the signer config but the context should take priority not config.

Reply:
Reminder: `toSet()` and `setOf()` return a `LinkedHashMap` which guarantees the order is preserved (minus any removed duplicate elements).